File: src/main/java/com/example/arttek/service/FirmService.java
package com.example.arttek.service;

import com.example.arttek.entity.Firm;
import com.example.arttek.repository.FirmRepository;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@AllArgsConstructor
public class FirmService {

    private final FirmRepository firmRepository;

    public Firm save(Firm firm) {
        if (firmRepository.findByAtiId(firm.getAtiId()).isPresent()) {
            throw new RuntimeException("Firm already exist with ati_id: " + firm.getAtiId());
        } else
            return firmRepository.save(firm);
    }
}

File: front-react/src/API/FirmService.js
import axios from "axios";

export default class FirmService {
    static async findByAtiId(atiId) {
        try {
            const response = await axios.get("http://localhost:8080/api/v1/firm/info?atiId=" + atiId)
            console.log(response.data)
            return response.data
        } catch (e) {
            throw e;
        }
    }

    static async save(firm) {
        const response = await axios.post("http://localhost:8080/api/v1/firm/", firm)
        return response.data
    }
}